### PR TITLE
Support all custom pull secret auth methods

### PR DESCRIPTION
**Pull request by gkrenn**
# Description

Currently only custom pull secrets with a username and a password are supported. 
This PR changes logic to use the docker client library to handle pull secrets in order to support all `dockerconfigjson` formats.

## How can this be tested?

- Create a custom pull secret for a private registry that does not use username and password (e.g. gcr)
- Set pull secret and image in the dynakube 
- The operator should be able to pull the image (visible in the logs)

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/controllers/csi/provisioner/agent.go [hunk 1]
```diff
@@ -98,17 +96,10 @@ func getUrlProperties(targetVersion, previousVersion string, pathResolver metada
...
+func newImageInstaller(ctx context.Context, fs afero.Fs, apiReader client.Reader, pathResolver metadata.PathResolver, db metadata.Access, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) { //nolint:revive // argument-limit doesn't apply to constructors
 	dockerConfig := dockerconfig.NewDockerConfig(apiReader, *dynakube)
 	if dynakube.Spec.CustomPullSecret != "" {
-		err := dockerConfig.SetupAuths(ctx)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if dynakube.Spec.TrustedCAs != "" {
```
Comment (luhi-DT) [inline src/controllers/csi/provisioner/agent.go:110]: You are now storing the trustedCAs only if there is a customPullSecret, that can lead to problems if a customer has trustedCAs but no customPullSecret

File: src/dockerconfig/docker_config.go [hunk 2]
```diff
@@ -12,82 +12,120 @@ import (
...
-		log.Info("failed to parse pull secret content", "dynakube", config.Dynakube.Name)
+
+	if err := saveFile(registryCredentials, fs, config.RegistryAuthPath); err != nil {
+		log.Info("failed to store registry credentials", "dynakube", config.Dynakube.Name)
 		return err
 	}
-	config.Auths = dockerAuths
+
+	if config.Dynakube.Spec.TrustedCAs != "" {
+		config.TrustedCertsPath = path.Join(TmpPath, CADir, config.Dynakube.Name)
```
Comment (luhi-DT) [inline src/dockerconfig/docker_config.go:55]: a new function for the whole trustedCA part would make it way cleaner

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: src/controllers/csi/provisioner/agent.go (hunk 1 above)
Comment (gkrenn) [reply src/controllers/csi/provisioner/agent.go:110]: You are right, this if should not exist here.

Added a unit test to check for this in the future.

File: src/dockerconfig/docker_config.go (hunk 2 above)
Comment (gkrenn) [reply src/dockerconfig/docker_config.go]: Created 2 new functions for trustedCAs and auths to make it clear what the code does

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**